### PR TITLE
PWGLF: Lnn Task

### PR DESCRIPTION
**Pull request by mapalhares**
Lnn Task with formatting corrections.

### CODE REVIEW

File: PWGLF/TableProducer/Nuspex/lnnRecoTask.cxx [hunk 1]
```diff
@@ -0,0 +1,623 @@
...
+      int chargeFactor = -1 + 2 * lnnCand.isMatter;
+      hdEdx3HSel->Fill(chargeFactor * lnnCand.mom3HTPC, h3track.tpcSignal());
+      hNsigma3HSel->Fill(chargeFactor * lnnCand.mom3HTPC, lnnCand.nSigma3H);
+
+      lnnCandidates.push_back(lnnCand);
+    }
+  }
+
+  // Monte Carlo information
+  void fillMCinfo(aod::McTrackLabels const& trackLabels, aod::McParticles const& particlesMC)
```
Comment (vkucera) [inline PWGLF/TableProducer/Nuspex/lnnRecoTask.cxx:422]: Please check that you only declare parameters that you use (in other functions too).
```suggestion
  void fillMCinfo(aod::McTrackLabels const& trackLabels, aod::McParticles const&)
```

File: PWGLF/TableProducer/Nuspex/lnnRecoTask.cxx [hunk 2]
```diff
@@ -0,0 +1,623 @@
...
+              if (!((mcTrackPos.pdgCode() == h3DauPdg && mcTrackNeg.pdgCode() == -211) || (mcTrackPos.pdgCode() == 211 && mcTrackNeg.pdgCode() == -1 * h3DauPdg))) // TODO: check warning for constant comparison
+                continue;
+              if (std::abs(posMother.pdgCode()) != lnnPdg)
+                continue;
+
+              // Checking primary and second vertex with MC simulations
+              auto posPrimVtx = array{posMother.vx(), posMother.vy(), posMother.vz()};
+              auto secVtx = array{mcTrackPos.vx(), mcTrackPos.vy(), mcTrackPos.vz()};
+              lnnCand.gMom = array{posMother.px(), posMother.py(), posMother.pz()};
+              lnnCand.gMom3H = mcTrackPos.pdgCode() == h3DauPdg ? array{mcTrackPos.px(), mcTrackPos.py(), mcTrackPos.pz()} : array{mcTrackNeg.px(), mcTrackNeg.py(), mcTrackNeg.pz()};
```
Comment (vkucera) [inline PWGLF/TableProducer/Nuspex/lnnRecoTask.cxx:446]: Use the existing `pVector` column in the table.

File: PWGLF/TableProducer/Nuspex/lnnRecoTask.cxx [hunk 3]
```diff
@@ -0,0 +1,623 @@
...
+    }
+
+    // now we fill only the signal candidates that were not reconstructed
+    for (auto& mcPart : particlesMC) {
+
+      if (std::abs(mcPart.pdgCode()) != lnnPdg)
+        continue;
+      std::array<float, 3> secVtx;
+      std::array<float, 3> primVtx = {mcPart.vx(), mcPart.vy(), mcPart.vz()};
+      std::array<float, 3> momMother = {mcPart.px(), mcPart.py(), mcPart.pz()};
```
Comment (vkucera) [inline PWGLF/TableProducer/Nuspex/lnnRecoTask.cxx:560]: Use `pVector()`.

File: PWGLF/TableProducer/Nuspex/lnnRecoTask.cxx [hunk 4]
```diff
@@ -0,0 +1,623 @@
...
+      std::array<float, 3> mom3H;
+      bool is3HFound = false;
+      for (auto& mcDaught : mcPart.daughters_as<aod::McParticles>()) {
+        if (std::abs(mcDaught.pdgCode()) == h3DauPdg) {
+          secVtx = {mcDaught.vx(), mcDaught.vy(), mcDaught.vz()};
+          mom3H = {mcDaught.px(), mcDaught.py(), mcDaught.pz()};
```
Comment (vkucera) [inline PWGLF/TableProducer/Nuspex/lnnRecoTask.cxx:566]: Use `pVector()`.

File: PWGLF/TableProducer/Nuspex/lnnRecoTask.cxx (hunk 3 above)
Comment (vkucera) [inline PWGLF/TableProducer/Nuspex/lnnRecoTask.cxx:560]:
```suggestion
      std::array<float, 3> momMother = mcPart.pVector();
```

File: PWGLF/TableProducer/Nuspex/lnnRecoTask.cxx (hunk 4 above)
Comment (vkucera) [inline PWGLF/TableProducer/Nuspex/lnnRecoTask.cxx:566]:
```suggestion
          mom3H = mcDaught.pVector();
```

**Review by vkucera — CHANGES_REQUESTED** [review]

File: PWGLF/TableProducer/Nuspex/lnnRecoTask.cxx [hunk 5]
```diff
@@ -0,0 +1,633 @@
...
+              if (!((mcTrackPos.pdgCode() == h3DauPdg && mcTrackNeg.pdgCode() == -211) || (mcTrackPos.pdgCode() == 211 && mcTrackNeg.pdgCode() == -1 * h3DauPdg))) // TODO: check warning for constant comparison
+                continue;
+              if (std::abs(posMother.pdgCode()) != lnnPdg)
+                continue;
+
+              // Checking primary and second vertex with MC simulations
+              auto posPrimVtx = array{posMother.vx(), posMother.vy(), posMother.vz()};
+              auto secVtx = array{mcTrackPos.vx(), mcTrackPos.vy(), mcTrackPos.vz()};
+              lnnCand.gMom = array{posMother.pVector()};
+              lnnCand.gMom3H = mcTrackPos.pdgCode() == h3DauPdg ? array{mcTrackPos.pVector()} : array{mcTrackNeg.pVector()};
```
Comment (vkucera) [inline PWGLF/TableProducer/Nuspex/lnnRecoTask.cxx:454]:
```suggestion
              lnnCand.gMom = posMother.pVector();
              lnnCand.gMom3H = mcTrackPos.pdgCode() == h3DauPdg ? mcTrackPos.pVector() : mcTrackNeg.pVector();
```
The value returned by `pVector` is already an array.

File: PWGLF/TableProducer/Nuspex/lnnRecoTask.cxx [hunk 6]
```diff
@@ -0,0 +1,633 @@
...
+    }
+
+    // now we fill only the signal candidates that were not reconstructed
+    for (auto& mcPart : particlesMC) {
+
+      if (std::abs(mcPart.pdgCode()) != lnnPdg)
+        continue;
+      std::array<float, 3> secVtx;
+      std::array<float, 3> primVtx = {mcPart.vx(), mcPart.vy(), mcPart.vz()};
+      std::array<float, 3> momMother = {mcPart.pVector()};
```
Comment (vkucera) [inline PWGLF/TableProducer/Nuspex/lnnRecoTask.cxx:570]:
```suggestion
      std::array<float, 3> momMother = mcPart.pVector();
```

File: PWGLF/TableProducer/Nuspex/lnnRecoTask.cxx [hunk 7]
```diff
@@ -0,0 +1,633 @@
...
+      std::array<float, 3> mom3H;
+      bool is3HFound = false;
+      for (auto& mcDaught : mcPart.daughters_as<aod::McParticles>()) {
+        if (std::abs(mcDaught.pdgCode()) == h3DauPdg) {
+          secVtx = {mcDaught.vx(), mcDaught.vy(), mcDaught.vz()};
+          mom3H = {mcDaught.pVector()};
```
Comment (vkucera) [inline PWGLF/TableProducer/Nuspex/lnnRecoTask.cxx:576]:
```suggestion
          mom3H = mcDaught.pVector();
```

**Review by vkucera — CHANGES_REQUESTED** [review]

**Comment by vkucera** [comment]
You don't need to fix the formatting by hand. Just merge this automatic PR: https://github.com/mapalhares/O2Physics/pull/3

**Comment by mapalhares** [comment]
Thank you, Vit! And I'm sorry about the mistakes, it's my first time doing it.

**Review by vkucera — APPROVED** [review]
Thanks @mapalhares for addressing my comments.

**Pull request merged**